shm/Game.cpp

```diff
@@ -0,0 +1,279 @@
+
+            std::cout << "Amount:\n";
+            limit = player.getMoney() / cargo->getPrice();
+            limit = std::min(limit, cargo->getAmount()) + 1;
+            size_t amount = ChoseOption(limit);
+            if (amount >= limit) {
+                return;
+            }
+            
+            Response response = store.sell(cargo, amount, &player);
+            printResponse(response);
+        }
+    }
+
+    std::string getName() const override { return "Buy"; }
+};
+
+class TradeSell : public Command {
+public:
+    TradeSell() = default;
+    void operator()(Player& player, Store& store) override {
+        while (true) {
+            size_t limit = 0;
+            printShipCargoList(player);
+            printPlayerStatus(player);
+
+            std::cout << "Which Cargo want to sell to store?\n";
+            limit = player.getShip()->getCargoVec().size();
+            auto chose = ChoseOption(limit);
+            if (chose >= limit) {
+                break;
+            }
+            const Cargo* const cargo = ChoseCargoFromShip(chose, player);
+
+            std::cout << "Amount:\n";
+            limit = player.getMoney() / cargo->getPrice();
+            limit = std::min(limit, cargo->getAmount()) + 1;
+            size_t amount = ChoseOption(limit);
+            if (amount >= limit) {
+                break;
+            }
+
+            Response response = store.buy(cargo, amount, &player);
+            printResponse(response);
+        }
+    }
+
+    std::string getName() const override { return "Sell"; }
+};
+
+class NextDay : public Command {
+public:
+    NextDay() = default;
+    void operator()([[maybe_unused]] Player& player, [[maybe_unused]] Store& store) override {
+        Time::getInstance()->nextDay();
+    }
+
+    std::string getName() const override { return "Next day"; }
+};
+
+class Travel : public Command {
+public:
+    Travel() = default;
+    void operator()([[maybe_unused]] Player& player, [[maybe_unused]] Store& store) override {
+        for (size_t day = 1; day <= distance; ++day) {
+            nextDay(player, store);
+            std::cout << "Day " << day << " on travel\n";
+        }
+    }
+
+    std::string getName() const override { return "Travel"; }
+
+private:
+    NextDay nextDay;
+    size_t distance = 5;//for show concept
+};
+
+void Game::startGame() {
+    init();
+    //auto time = Time::getInstance();
+
+    unsigned char letter;
+    while (!endGame and !playerWin) {
+        printStoreCargoList(store);
+        printShipCargoList(player);
+
+        printPlayerStatus(player);
+
+        printMenu(commands);
+        std::cin >> letter;
+        //std::cin.clear();
+
+        size_t optionId = static_cast<size_t>(letter - '0');
+        if (optionId < commands.size()) {
+            Command& command = *commands[optionId];
+            command(player, store);
+        } else if (optionId == commands.size()) {
+            endGame = true;
+        }
```

Comment:
I think map will be better. Then you will call commands[letter] and avoid these few lines. Also, make exit game a command.